core/server/master/src/main/java/alluxio/master/file/meta/InodeTreePersistentState.java

```diff
@@ -412,6 +412,8 @@ private void applyUpdateInode(UpdateInodeEntry entry) {
           for (String medium : entry.getMediumTypeList()) {
             if (mediaList.contains(medium)) {
               file.getMediumTypes().add(medium);
+            } else {
+              LOG.warn("pinned medium types contain invalid medium");
```

Comment:
Logging this on the master is not that useful, since it can't be mapped to the client request. I would prefer to do a client side check